### PR TITLE
feat: surfacing potential duplicates

### PR DESCRIPTION
**Pull request by gijoe0295**
### Details

This PR implements UI for duplicated transaction violations.

### Fixed Issues

$ https://github.com/Expensify/App/issues/32607
PROPOSAL: https://github.com/Expensify/App/issues/32607#issuecomment-2049858277

### Tests

> [!NOTE]  
> Enable `Use Staging Server` if testing on dev
> The Collect workspace has Expenses >> Expense Violations enabled

1. Log in with Collect workspace's member account
2. Create 2 expenses with same AMOUNT, CURRENCY, MERCHANT and DATE (select one in the past) to the Collect workspace
4. Go to the workspace chat in which the expense report should be included.
5. Verify in the workspace chat, the report preview has a red dot indicator
6. Go to the expense report
7. Verify each expense preview has a red dot indicator and `Cash · Duplicate · Hold` on top
8. Go to the transaction report
9. Verify the `HOLD` banner is added with the following copy `This expense has the same details as another one. Review the duplicates to remove the hold.`
10. Verify a green button `Review duplicates` is in the request header (pressing it does nothing)

- [x] Verify that no errors appear in the JS console

### Offline tests

NA

### QA Steps

> [!NOTE]  
> The Collect workspace has Expenses >> Expense Violations enabled

1. Log in with Collect workspace's member account
2. Create 2 expenses with same AMOUNT, CURRENCY, MERCHANT and DATE to the Collect workspace
3. Go to the transaction report
4. Go to the policy expense report
5. Verify in the policy expense report, the report preview has a red dot indicator
6. Go to the expense report
7. Verify each expense preview has a red dot indicator and `Cash · Duplicate · Hold` on top
8. Go to the transaction report
9. Verify the `HOLD` banner is added with the following copy `This expense has the same details as another one. Review the duplicates to remove the hold.`
10. Verify a green button `Review duplicates` is in the request header (pressing it does nothing)

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot_1714075213
Screenshot_1714075183
Screenshot_1714075207
Screenshot_1714075221

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<--- add screenshots or videos here -->

Screenshot_1714074129
Screenshot_1714074192
Screenshot_1714074178
Screenshot_1714074181

</details>

<details>
<summary>iOS: Native</summary>

Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 33 10
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 33 00
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 33 03
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 33 17

</details>

<details>
<summary>iOS: mWeb Safari</summary>

Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 35 37
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 35 30
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 35 39
Simulator Screenshot - iPhone 15 Pro Max - 2024-04-26 at 02 35 34

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

Screenshot 2024-04-26 at 00 38 28
Screenshot 2024-04-26 at 00 38 37
Screenshot 2024-04-26 at 02 12 15
Screenshot 2024-04-26 at 02 15 06

</details>

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2024-04-26 at 02 26 16
Screenshot 2024-04-26 at 02 25 51
Screenshot 2024-04-26 at 02 23 09

</details>

### CODE REVIEW

**Comment by akinwale** [comment]
@gijoe0295 Please type include your test steps as text in the OP instead of using an image.

File: src/languages/es.ts [hunk 1]
```diff
@@ -3379,7 +3381,7 @@ export default {
         categoryOutOfPolicy: 'La categoría ya no es válida',
         conversionSurcharge: ({surcharge}: ViolationsConversionSurchargeParams = {}) => `${surcharge}% de recargo aplicado`,
         customUnitOutOfPolicy: 'La unidad ya no es válida',
-        duplicatedTransaction: 'Posible duplicado',
+        duplicatedTransaction: 'Duplicada',
```
Comment (pecanoro) [inline src/languages/es.ts:3384]:
```suggestion
        duplicatedTransaction: 'Duplicado',
```

File: src/components/HoldBanner.tsx [hunk 2]
```diff
@@ -5,14 +5,19 @@ import useThemeStyles from '@hooks/useThemeStyles';
 import Text from './Text';
 import TextPill from './TextPill';
 
-function HoldBanner() {
+type HoldBannerProps = {
+    isRequestDuplicate?: boolean;
+    shouldShowBorderBottom?: boolean;
```
Comment (pecanoro) [inline src/components/HoldBanner.tsx:10]: Don't forget to add descriptions for the props

**Review by pecanoro — CHANGES_REQUESTED** [review]

**Comment by akinwale** [comment]
I'll review later today / tomorrow.

**Comment by trjExpensify** [comment]
@akinwale did you review?

**Comment by akinwale** [comment]
> @akinwale did you review?

I didn't finish yet. Had some issues with the Android build which caused some delays. I will wrap up the review in a few hours.

**Comment by akinwale** [comment]
@gijoe0295 I'm not seeing the duplicate violation error come up.

I have the following set:
* `Use staging server` is on
* `canUseViolations` beta returns true in Permissions
* The test account: akinwale+expensify83@gmail.com also has violations enabled in BE

Is there anything I might be missing?

**Comment by pecanoro** [comment]
Sorry, I have been sick, let me check! So the `transactionViolations_` onyx key does not contain any violations for those transactions? Is what you are showing in the image the whole only key? For two transactions to be marked as duplicated, they need to have the same currency, amount and date.

**Comment by pecanoro** [comment]
Are they in a free wrokspace by any chance? They need to be in a collect workspace

**Comment by pecanoro** [comment]
Ohh I checked the workspace, it's probably because violations are not enabled, you need to change this setting:

image

**Comment by akinwale** [comment]
> Ohh I checked the workspace, it's probably because violations are not enabled, you need to change this setting:
> 

@pecanoro I had this enabled but I got the same behaviour. I'll try to obtain the transaction IDs to post in a couple of hours.

**Comment by pecanoro** [comment]
Hmm, TypeScript Checks seems to be failing

**Comment by pecanoro** [comment]
@akinwale Friendly bump!

**Comment by akinwale** [comment]
Sorry, didn't realise I was delaying this. I will run the test again today.

**Comment by trjExpensify** [comment]
@akinwale can you give us an update on this please?

**Comment by akinwale** [comment]
I got sidetracked and didn't finish up with testing, and I was travelling over the past couple of days. I will post another update later today after I run the tests.

**Comment by akinwale** [comment]
@pecanoro @gijoe0295 

I tested again, and the duplicate violation still didn't show up. I have included a video below.

The transaction IDs are:
* `3945434075758664250`
* `4079000804413681565`

https://github.com/Expensify/App/assets/4160319/3f0ffffe-7537-4fd0-90ad-f649f8f521ec

**Comment by pecanoro** [comment]
@akinwale, What's the email again you are using? I see you are calling it from `akinwale+expensify48@gmail.com` but that email is not in the beta (akinwale+expensify83@gmail.com is on it though). I added that email as well, please try in 30 min again. 

These three emails should be there:

image

**Comment by akinwale** [comment]
@pecanoro akinwale+expensify48@gmail.com is the workspace owner, while akinwale+expensify83@gmail.com is the employee account that I am submitting the expenses from. I'll retest again later today.

**Comment by trjExpensify** [comment]
@gijoe0295 there's conflicts now, can you fix them please?

**Comment by trjExpensify** [comment]
@akinwale how did you get on with the retests?

**Comment by pecanoro** [comment]
@akinwale Friendly bump!

**Comment by akinwale** [comment]
@pecanoro I just tested again, but still no joy here. :(

The corresponding transaction IDs are:
* `8727876317130291128`
* `4648699694032853676`

Video included below.

https://github.com/Expensify/App/assets/4160319/8b463820-fd65-4721-ac31-adf791ae4877

**Comment by pecanoro** [comment]
@akinwale Does the violation show in the report in oldDot? Let's start there haha

**Comment by akinwale** [comment]
Well, no it doesn't show up. The text for the Notes displayed by the yellow warning is "Cash expense with no receipt".

Screenshot 2024-05-31 at 01 47 57

**Comment by akinwale** [comment]
@pecanoro Apologies, forgot to tag you in my previous reply: https://github.com/Expensify/App/pull/40153#issuecomment-2141056867

**Comment by akinwale** [comment]
> @akinwale when you enable violations, if you refresh the page, does it show as disabled again? I think we might have broken that switch internally. It's resetting for me to disabled

@pecanoro Yes, I can confirm this behaviour.

https://github.com/Expensify/App/assets/4160319/f39b996e-e31e-48f0-8e88-3a1bf34c62ed

**Comment by pecanoro** [comment]
It's a weird behavior, but I think we recently changed violations to be a control feature. Can you try upgrading the workspace to control and see if the violations show up?

**Comment by pecanoro** [comment]
Hmm, actually they should show on collect, but they didn't show for me until I upgraded, even if I downgraded later. Not sure, I will check tomorrow. For now, let me know if upgrading works

**Comment by akinwale** [comment]
@pecanoro So I was able to update to Control and the violations for receipt required showed up (>=$25). However, after enabling the Staging Server option and creating $15 expenses with the same merchant, the duplicate violation still wasn't displayed.

The transaction IDs are:
* `3683644982551534836`
* `5120529863938916787`

https://github.com/Expensify/App/assets/4160319/eb5ba66f-4e2d-4265-bde5-834a868d18fb

**Comment by pecanoro** [comment]
@akinwale Did you click on the transaction thread after creating the request? I think we only fetch violations when we call OpenReport

**Comment by akinwale** [comment]
> In the logs, I can't find any trace of computing the violations at all or opening the transaction report, that's why I am so confused. @akinwale Can you try again and give me the transactionIDs and the reportID? I am going to check again

@pecanoro Do I have to create completely new transactions or would the existing transactions work?

**Comment by pecanoro** [comment]
@akinwale I pinged you [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1717514440651929) so we can resolve it much faster on Slack!

**Comment by akinwale** [comment]
@gijoe0295 Please add a note to your test step that the user needs to open the transaction thread before the violations are actually displayed.

**Comment by akinwale** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Review by akinwale — APPROVED** [review]
LGTM.

**Comment by trjExpensify** [comment]
Wahoo, progress! @gijoe0295 we have a bunch of conflicts here now. Can you address them?

**Review by pecanoro — APPROVED** [review]
Looks great! I am going to test it really quick and I will merge

**Comment by pecanoro** [comment]
Seems to be working well, merging!

**Comment by chiragsalian** [comment]
Just letting you know the PR here had a minor regression. Full details about the regression is posted in the CP PR [here](https://github.com/Expensify/App/pull/43446).

File: src/libs/TransactionUtils.ts [hunk 3]
```diff
@@ -631,7 +658,7 @@ function isOnHold(transaction: OnyxEntry<Transaction>): boolean {
         return false;
     }
 
-    return !!transaction.comment?.hold;
+    return !!transaction.comment?.hold || isDuplicate(transaction.transactionID, true);
 }
```
Comment (ahmedGaber93) [inline src/libs/TransactionUtils.ts:662]: add `isDuplicate` to condition here cause normal "Unhold" option shows for duplicated expense. and we handled it here https://github.com/Expensify/App/issues/44892

**Comment by sobitneupane** [comment]
[This case](https://github.com/Expensify/App/issues/49793) was missed during the feature implementation. For one-expense reports, the Review Duplicate button is not displayed.

File: src/libs/TransactionUtils.ts [hunk 4]
```diff
@@ -623,6 +633,23 @@ function getRecentTransactions(transactions: Record<string, string>, size = 2):
...
+ */
+function isDuplicate(transactionID: string, checkDismissed = false): boolean {
+    const hasDuplicatedViolation = !!allTransactionViolations?.[`${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`]?.some(
+        (violation: TransactionViolation) => violation.name === CONST.VIOLATIONS.DUPLICATED_TRANSACTION,
+    );
+    if (!checkDismissed) {
+        return hasDuplicatedViolation;
+    }
+    const didDismissedViolation =
+        allTransactions?.[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`]?.comment?.dismissedViolations?.duplicatedTransaction?.[currentUserEmail] === `${currentUserAccountID}`;
```
Comment (ahmedGaber93) [inline src/libs/TransactionUtils.ts:649]: Coming from https://github.com/Expensify/App/issues/66181, 

From the OP "Feature Description" screenshots, BE is returning the `duplicatedTransaction` value as a `[currentUserEmail]: timestamp`, but in the PR it was implemented as `[currentUserEmail]: currentUserAccountID`.

`1711660500` below is a timestamp (Thursday, March 28, 2024 9:15:00 PM).

Image

**Pull request merged**